source/nodejs/tests/unit-tests/src/unit-tests.test.ts

```diff
@@ -14,3 +17,123 @@ describe("test Text property on TextBlock", () => {
         expect('{"type":"TextBlock","text":"Some text"}').toBe(JSON.stringify(textBlockJson));
     });
 });
+
+describe("Test Templating Library", () => {
+	it("BasicTemplate", () => {
+		const templatePayload = {
+			"type": "AdaptiveCard",
+			"version": "1.0",
+			"body": [
+				{
+					"type": "TextBlock",
+					"text": "Hello ${name}!"
+				}
+			]
+		};
+		
+		let template = new ACData.Template(templatePayload);
+		
+		let context: ACData.IEvaluationContext = {
+			$root: {
+				"name": "Adaptive Cards"
+			}
+		};
+		
+		let card = template.expand(context);
+
+		const expectedOutput = {
+			"type": "AdaptiveCard",
+			"version": "1.0",
+			"body": [ 
+				{ 
+					"type": "TextBlock", 
+					"text": "Hello Adaptive Cards!" 
+				} 
+			]
+		}
+
+		expect(card).toStrictEqual(expectedOutput);
+	})
+
+	it("BasicTemplateWithHost", () => {
+		const templatePayload = {
+			"type": "AdaptiveCard",
+			"version": "1.0",
+			"body": [
+				{
+					"type": "TextBlock",
+					"text": "Hello ${name}! You're using the ${$host.WindowsTheme} theme!"
+				}
+			]
+		};
+		
+		let template = new ACData.Template(templatePayload);
+		
+		let context: ACData.IEvaluationContext = {
+			$root: {
+				"name": "Adaptive Cards"
+			},
+			$host: {
+				"WindowsTheme": "Light"
+			}
+		};
+		
+		let card = template.expand(context);
+
+		const expectedOutput = {
+			"type": "AdaptiveCard",
+			"version": "1.0",
+			"body": [ 
+				{ 
+					"type": "TextBlock", 
+					"text": "Hello Adaptive Cards! You're using the Light theme!" 
+				} 
+			]
+		};
+
+		expect(card).toStrictEqual(expectedOutput);
+	});
+
+	it("ComplexTemplate", () => {
+		const templateObject = fs.readFileSync("template-test-resources/complex-template.json", "utf8");
```

Comment:
Nit: It's better to refactor the test cases and remove duplication.

For example
```
const testData = [{template: "complex-template.json", expect: "complex-template.output.json"},  ];

for (test of testData) {
    runTest(test);
}
```

or for input/output from files 
```
runTest(templateFile, outputFile....)
```

or even more generic:
```
runTest(loadFile(templateFile), loadFile(outputFile)....)
```